azurerm/resource_arm_storage_management_policy.go

```diff
@@ -0,0 +1,389 @@
+package azurerm
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/hashicorp/terraform/helper/validation"
+
+	"github.com/Azure/azure-sdk-for-go/services/storage/mgmt/2019-04-01/storage"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceArmStorageManagementPolicy() *schema.Resource {
+	return &schema.Resource{
+		Create:        resourceArmStorageManagementPolicyCreateOrUpdate,
+		Read:          resourceArmStorageManagementPolicyRead,
+		Update:        resourceArmStorageManagementPolicyCreateOrUpdate,
+		Delete:        resourceArmStorageManagementPolicyDelete,
+		SchemaVersion: 1,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"storage_account_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"rule": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringMatch(
+								regexp.MustCompile(`^[a-zA-Z0-9]*$`),
+								"A rule name can contain any combination of alpha numeric characters.",
+							),
+						},
+						"enabled": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+						"type": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice([]string{"Lifecycle"}, false),
```

Comment:
Since `type` is required and `Lifecycle` is the only accepted value. We can remove it from the schema and just reference `Lifecycle` when we create the rule below.